modules/ROOT/pages/dataweave-streaming.adoc

```diff
@@ -0,0 +1,398 @@
+
+* `streaming` property for reading source data as a stream
+* `deferred` writer property for passing an output stream directly to the next message processor in a flow
+
+To enable streaming when reading source data, you must set the `streaming` reader property to `true` on the data source. You can enable streaming on a data source through the *MIME Type* setting `outputMimeType`. You can set the property in a Mule component that accepts source data, such as an HTTP listener or an On New or Updated File operation.
+
+[source,XML,linenums]
+----
+<flow name="dw-streaming-example" >
+  <http:listener doc:name="Listener"
+      outputMimeType="application/json; streaming=true"
+      config-ref="HTTP_Listener_config" path="/input"/>
+</flow>
+----
+
+Notice that `streaming=true` is part of the `outputMimeType` value. Other Mule components, such as the File and FTP components, also support the MIME Type setting.
+
+To pass the streamed output to the next message processor, you can use the `output` directive in a DataWeave script, for example:
+
+[source,XML,linenums]
+----
+output application/json deferred=true
+----
+
+For more detail on use of the `deferred` property, see <<stream_output>>.
+
+[[stream_csv]]
+== Streaming CSV
+
+CSV is simplest format for streaming because of its structure. Each row below the CSV header is a streamable record. The following CSV example consists of records that contain `name`, `lastName`, and `age` values.
+​
+[source,CSV,linenums]
+----
+name,lastName,age
+mariano,achaval,37
+leandro,shokida,30
+pedro,achaval,4
+christian,chibana,25
+sara,achaval,2
+matias,achaval,8
+----
+​
+To stream the CSV example, the following script selects values from each record. It uses the `map` function to iterate over each record in the document.
+​
+[source,DataWeave,linenums]
+----
+payload map (record) ->
+    {
+        FullName: record.lastName ++ "," ++ record.name,
+        Age: record.age
+    }
+----
+
+Although streaming does not support random access to the entire document, a DataWeave script can access data randomly _within each record_ because each record is loaded into memory. For example, the expression `record.lastName ++ "," ++ record.name,` can access a `lastName` value before it accesses a `name` value even though the order of values is reversed in the input.
+
+However, streaming does not work in the following script. The script requires random access to the entire document to return the elements in a different order than they are given in the input.
+​
+[source,DataWeave,linenums]
+----
+[payload[-2], payload[-1], payload[3]]
+----
+​
+[[stream_json]]
+== Streaming JSON
+​
+The unit of a JSON stream is each element in an array.
+
+Note that DataWeave added limited support for JSON streaming in version 2.2.0 for Mule 4.2. To stream JSON in that version, DataWeave required the root to be an array. In Mule 4.3, DataWeave extended support for streaming to arrays that are not at the root of the input.
```

Comment:
Consider rewording to focus on evolution Mule 4.2 to 4.3: 

```suggestion
Note that DataWeave 2.2.0 support for JSON streaming for Mule 4.2 was limited by the requirement that the root be an array. DataWeave support in Mule 4.3 includes streaming to arrays that are not at the root of the input.
```

Reply:
Provide a brief text intro. to the example? It does not logically follow the previous note.
Or maybe move it to follow the intro, and then follow the example with the note.